sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_limits.py

```diff
@@ -43,6 +44,8 @@ def __init__(self, *, timeout: Optional[int] = None):
         self.timeout = timeout
 
     def _to_rest_object(self) -> RestCommandJobLimits:
+        if isinstance(self.timeout, PipelineInput):
+            return RestCommandJobLimits(timeout=self.timeout)
```

Comment:
why not change the _from_rest_object also?

Reply:
Original steps are:
```python
      rest_limits = RestCommandJobLimits.from_dict(obj["limits"])
      obj["limits"] = CommandJobLimits()._from_rest_object(rest_limits)
```

When getting response from server (obj["limits"]), the str-to-iso-duration will happen in the first line. So change "CommandJobLimits._from_rest_object" will not help.

Reply:
Good point! Updated the code, please take a look.